docs/examples/junit4/generic/src/test/java/generic/CmdModifierTest.java

```diff
@@ -18,10 +18,13 @@
     // }
 
     // memory {
+    private HostConfig hostConfig = HostConfig.newHostConfig()
+        .withMemory((long) 8 * 1024 * 1024)
+        .withMemorySwap((long) 12 * 1024 * 1024);
+
     @Rule
     public GenericContainer memoryLimitedRedis = new GenericContainer<>("redis:3.0.2")
-            .withCreateContainerCmdModifier(cmd -> cmd.withMemory((long) 8 * 1024 * 1024))
-            .withCreateContainerCmdModifier(cmd -> cmd.withMemorySwap((long) 12 * 1024 * 1024));
+            .withCreateContainerCmdModifier(cmd -> cmd.withHostConfig(hostConfig);
```

Comment:
this is incorrect.

It should be `-> cmd.withHostConfig(cmd.getHostConfig().withMemory((long) 8 * 1024 * 1024))`-like.

Otherwise, you're overwriting the previously set host values